tests/test_text.py

```diff
@@ -0,0 +1,464 @@
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        method = SelectionMethod.TextBased(num_features=None,
+                                           optimization_method="random",
+                                           cost_metric="unicost",
+                                           trials=5)
+
+        selector = Selective(method)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        # Check whether the selector.transform() is returned a DataFrame or not
+        self.assertTrue(isinstance(selected_features, pd.DataFrame))
+
+        # Check whether the number of row `selected_features' is the same as
+        # the number of rows in the input data or not
+        self.assertEqual(selected_features.shape[0], data.shape[0])
+
+        # Check whether the number of columns in `selected_features' is equal to num_features (user input) or not
+        if method.num_features is None:
+            pass
+        else:
+            self.assertEqual(selected_features.shape[1], 2)
+
+    def test_text_based_random_diverse(self):
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        # "diverse" is Default cost metric
+        method = SelectionMethod.TextBased(num_features=None,
+                                           featurization_method=TextWiser(Embedding.TfIdf(min_df=0),
+                                                                          [Transformation.NMF(n_components=20),
+                                                                           Transformation.SVD(n_components=10)]),
+                                           optimization_method="random",
+                                           trials=5)
+
+        selector = Selective(method)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        self.assertTrue(isinstance(selected_features, pd.DataFrame))
+        self.assertEqual(selected_features.shape[0], data.shape[0])
+        if method.num_features is None:
```

Comment:
I think this test also just pass-es and not verifies

Reply:
New asserts added to verify the selection